### ISSUE TITLE
A question on Encoder cells and Decoder cells

### ISSUE THREAD

**Issue by KavinduZoysa** [una-dinosauria/human-motion-prediction#42]
Hi @una-dinosauria,

I have a question on modeling Encoder cells and Decoder cells which still I was unable to solve. Let me explain it like this.

When the seq2seq model is created, first it will create the GRUcell like below.
https://github.com/una-dinosauria/human-motion-prediction/blob/c9a2774fe59bb78c6ec6d04e15f2cb7243c4348c/src/seq2seq_model.py#L83

Then it will add the Linear Space decorder to the GRU cell like below.
https://github.com/una-dinosauria/human-motion-prediction/blob/c9a2774fe59bb78c6ec6d04e15f2cb7243c4348c/src/seq2seq_model.py#L112

After the it will add the redual wrapper to model velocity as below.
https://github.com/una-dinosauria/human-motion-prediction/blob/c9a2774fe59bb78c6ec6d04e15f2cb7243c4348c/src/seq2seq_model.py#L116

Finally that cell is used to model full seq2seq model like below.
https://github.com/una-dinosauria/human-motion-prediction/blob/c9a2774fe59bb78c6ec6d04e15f2cb7243c4348c/src/seq2seq_model.py#L138
 
According to the below image, the cell which is wrapped by Linear Space decorder wrapper and redual wrapper is only used in decorder and in the encorder we have a simple cell. 
![screenshot from 2018-10-02 07-09-35](https://user-images.githubusercontent.com/28332708/46324642-265adc80-c612-11e8-9836-d7d0fe181e91.png)

My problem is, according to this implimentation do we expect to run Residual wrapper and Linear space decorder wrapper in side the both encorder and decorder?
Highly appreciated you inputs.

Thank you,
Kavindu

**Comment by una-dinosauria** [comment]
Hi @KavinduZoysa,

In tensorflow, one builds a computational graph and then asks for a node to be evaluated given certain inputs. Because tf knows the graph in advance, it will simply ignore nodes that are not necessary to evaluate the node that one requests (usually the loss function).

I think this is the case for the extra outputs of the encoder -- they do not affect the loss, so they should not be evaluated.

I hope that clarifies things.

**Comment by KavinduZoysa** [comment]
Hi @una-dinosauria,

Yes, I understood. Thank you for the answer. 

In my MSc programme, I am using this paper to complete one of my projects. I am trying to reproduce the results and I am trying to improve it by changing the architecture/algorithm. That's why I asked you several questions previously to understand the code properly. Thank you again for giving me answers to those questions as well.

Thank you,
Kavindu
